src/editor/Core/Types.re

```diff
@@ -155,6 +155,10 @@ module BufferNotification = {
     bufferId: int,
     buffers: list(BufferMetadata.t),
   };
+
+  let getBufferMetadata = (id, v: t) => {
+    v.buffers |> List.filter((b: BufferMetadata.t) => b.id == id) |> List.hd;
```

Comment:
@bryphe `hd` can throw a `Failure` which might cause a runtime collapse we could wrap this in a try or not sure if theres an alternative that returns an `option`

Reply:
Good call - I refactored to use `List.nth_opt` instead 👍